lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_controller.dart

```diff
@@ -0,0 +1,215 @@
+    try {
+      _getAllMailboxInteractor = Get.find<GetAllMailboxInteractor>();
+      _refreshAllMailboxInteractor = Get.find<RefreshAllMailboxInteractor>();
+      _subscribeMailboxInteractor = Get.find<SubscribeMailboxInteractor>();
+    } catch (e) {
+      logError('MailboxVisibilityController::onInit(): ${e.toString()}');
+    }
+  }
+
+  @override
+  void onDone() {
+    viewState.value.fold((failure) {}, (success) {
+      if (success is GetAllMailboxSuccess) {
+        _currentMailboxState = success.currentMailboxState;
+        _buildMailboxTreeHasSubscribed(success.mailboxList);
+      } else if (success is RefreshChangesAllMailboxSuccess) {
+        _currentMailboxState = success.currentMailboxState;
+        _refreshMailboxTreeHasSubscribed(success.mailboxList);
+      } else if (success is SubscribeMailboxSuccess) {
+        _subscribeMailboxSuccess(success);
+      }
+    });
+
+  }
+
+  @override
+  void onReady() {
+    final _session = _accountDashBoardController.sessionCurrent.value;
+    final _accountId = _accountDashBoardController.accountId.value;
+    if(_session != null && _accountId != null) {
+      getAllMailboxAction(_session, _accountId);
+    }
+    _initCollapseMailboxCategories();
+    super.onReady();
+  }
+
+  void _initCollapseMailboxCategories() {
+    if (kIsWeb &&
+        currentContext != null &&
+        (_responsiveUtils.isMobile(currentContext!) ||
+            _responsiveUtils.isTablet(currentContext!))) {
+      mailboxCategoriesExpandMode.value = MailboxCategoriesExpandMode(
+          defaultMailbox: ExpandMode.COLLAPSE,
+          personalMailboxes: ExpandMode.COLLAPSE,
+          teamMailboxes: ExpandMode.COLLAPSE);
+    } else {
+      mailboxCategoriesExpandMode.value = MailboxCategoriesExpandMode.initial();
+    }
+  }
+
+  void getAllMailboxAction(Session session, AccountId accountId) async {
+    if(_getAllMailboxInteractor != null){
+      consumeState(_getAllMailboxInteractor!.execute(session, accountId));
+    }
+  }
+
+  void refreshAllMailbox() {
+    final session = _accountDashBoardController.sessionCurrent.value;
+    final accountId = _accountDashBoardController.accountId.value;
+    if (session != null && accountId != null) {
+      consumeState(_getAllMailboxInteractor!.execute(session, accountId));
+    }
+  }
+
+  void refreshMailboxChanges({jmap.State? currentMailboxState}) {
+    log('MailboxVisibilityController::refreshMailboxChanges(): currentMailboxState: $currentMailboxState');
+    final newMailboxState = currentMailboxState ?? _currentMailboxState;
+    log('MailboxVisibilityController::refreshMailboxChanges(): newMailboxState: $newMailboxState');
+    final session = _accountDashBoardController.sessionCurrent.value;
+    final accountId = _accountDashBoardController.accountId.value;
+    if (accountId != null && session != null && newMailboxState != null) {
+      consumeState(_refreshAllMailboxInteractor!.execute(session, accountId, newMailboxState));
+    }
+  }
```

Comment:
IMO, it is duplicated with other mailbox controller? If yes, why not define it in `base` class?

Reply:
func `refreshAllMailbox ()` is not in use I will delete it and i use only `refreshMailboxChanges()`